### ISSUE TITLE
Getting the word-piece tokenized text for USE

### ISSUE THREAD

**Issue by GeetDsa** [tensorflow/hub#686]
The pipeline of Universal Sentence Encoder (USE), first tokenizes the text, then obtaining the embeddings, and passing these through set of layers across the transformers and finally providing the embeddings for the given sentence.

Is it possible to get the tokenized text (or list of tokens that the input sentence is split into). (Similar to how BERT provides its tokenizer as well as the pre-trained model)

**Comment by jaxlaw** [comment]
Please see https://github.com/tensorflow/hub/issues/662

**Comment by arghyaganguly** [comment]
@GeetDsa ,please confirm whether the workflow in #662 suits your requirement.Thanks.

**Comment by VibhuJawa** [comment]
@arghyaganguly , 

I dont think the workaround in  the issue #662 works for `USE ` . It seems like it only works for `MUSE`.   I tried with both `sentencepiecetokenizeop` , `wordpiecetokenizeop` or even just `tokenizeop` but cant seem to find the right function. 


```python
import importlib
MODEL_PATH = "./tf_model/" # just downloaded and extracted in this dir
loader_impl = importlib.import_module('tensorflow.python.saved_model.loader_impl')

saved_model = loader_impl.parse_saved_model(MODEL_PATH)
graph = saved_model.meta_graphs[0].graph_def


# extract functions that contain SentencePiece somewhere in there
fns = [f for f in saved_model.meta_graphs[0].graph_def.library.function if "sentencepiecetokenizeop" in str(f).lower()];
print(fns)
``` 
```
[]
```

Can you please re-open this. issue .

**Comment by arghyaganguly** [comment]
@VibhuJawa , tokenzers : sentencepiecetokenizeop , wordpiecetokenizeop are used by MUSE  while in case of USE, an internal mechanism is used.

**Comment by VibhuJawa** [comment]
> @VibhuJawa , tokenzers : sentencepiecetokenizeop , wordpiecetokenizeop are used by MUSE while in case of USE, an internal mechanism is used.

Is there a way to access the internal mechanism  ?   Even a link to the code will help a lot .  

For context: I am trying to use `[RAPIDS](https://rapids.ai/)` for string processing and want to use that library for tokenizing. Then i want to pass those tokenized results to `Tensorflow's` USE model .

**Comment by jaxlaw** [comment]
Unfortunately the text preprocessing of this particular version is not publicly available. But the gist of it is simply a string split by space (with tf.string_split) and some regex to normalize the tokens.

**Comment by UsharaniPagadala** [comment]
@GeetDsa 
Please respond to the above comment and confirm if this issue can be closed.Thanks
